File: src/main/java/com/vaadin/demo/component/datepicker/DatePickerInternationalization.java
package com.vaadin.demo.component.datepicker;

import java.util.List;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.datepicker.DatePicker;
import com.vaadin.flow.router.Route;

@Route("date-picker-internationalization")
public class DatePickerInternationalization extends Div {

    public DatePickerInternationalization() {
        DatePicker datePicker = new DatePicker("Sitzungsdatum");

        // tag::snippet[]
        DatePicker.DatePickerI18n germanI18n = new DatePicker.DatePickerI18n();
        germanI18n.setMonthNames(List.of("Januar", "Februar", "März", "April",
                "Mai", "Juni", "Juli", "August", "September", "Oktober",
                "November", "Dezember"));
        germanI18n.setWeekdays(List.of("Sonntag", "Montag", "Dienstag",
                "Mittwoch", "Donnerstag", "Freitag", "Samstag"));
        germanI18n.setWeekdaysShort(
                List.of("So", "Mo", "Di", "Mi", "Do", "Fr", "Sa"));
        germanI18n.setToday("Heute");
        germanI18n.setCancel("Abbrechen");

        datePicker.setI18n(germanI18n);
        // end::snippet[]

        add(datePicker);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<DatePickerInternationalization> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/datepicker/date-picker-internationalization.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, query } from 'lit/decorators.js';
import '@vaadin/date-picker';
import type { DatePicker } from '@vaadin/date-picker';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('date-picker-internationalization')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  // tag::snippet[]
  @query('vaadin-date-picker')
  private accessor datePicker!: DatePicker;

  protected override firstUpdated() {
    this.datePicker.i18n = {
      ...this.datePicker.i18n,
      monthNames: [
        'Januar',
        'Februar',
        'März',
        'April',
        'Mai',
        'Juni',
        'Juli',
        'August',
        'September',
        'Oktober',
        'November',
        'Dezember',
      ],
      weekdays: ['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag'],
      weekdaysShort: ['So', 'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa'],
      today: 'Heute',
      cancel: 'Abbrechen',
    };
  }

  protected override render() {
    return html`<vaadin-date-picker label="Sitzungsdatum"></vaadin-date-picker>`;
  }
  // end::snippet[]
}
